pkg/plugin/loader/loader.go

```diff
@@ -53,98 +50,139 @@ func LoadAllPlugins(namespace string, searchPath []string, selections []plugin.S
-func loadPlugin(namespace string, dfn plugin.Definition, masterAddress string) (plugin.Interface, error) {
-	// TODO(chuckha): We don't use the cfg for anything except passing a string around. Consider removing this struct.
-	cfg := &plugin.WorkerConfig{}
-	logrus.Infof("Loading plugin driver %v", dfn.Driver)
-	switch dfn.Driver {
-	case "DaemonSet":
-		cfg.MasterURL = "http://" + masterAddress + "/api/v1/results/by-node"
-		return daemonset.NewPlugin(namespace, dfn, cfg), nil
-	case "Job":
-		cfg.MasterURL = "http://" + masterAddress + "/api/v1/results/global"
-		return job.NewPlugin(namespace, dfn, cfg), nil
-	default:
-		return nil, errors.Errorf("Unknown driver %v", dfn.Driver)
+	pluginDefinitions = filterPluginDef(pluginDefinitions, selections)
+
+	plugins := []plugin.Interface{}
+	for _, def := range pluginDefinitions {
+		loadedPlugin, err := loadPlugin(def, namespace)
+		if err != nil {
+			return nil, errors.Wrapf(err, "couldn't load plugin %v", def.SonobuoyConfig.PluginName)
+		}
+		plugins = append(plugins, loadedPlugin)
 	}
-}
 
-// scanPlugins looks for Plugin Definition files in the given directory,
-// and returns an array of PluginDefinition structs.
-func scanPlugins(dir string) ([]plugin.Definition, error) {
-	var pluginDfns []plugin.Definition
+	return plugins, nil
+}
 
-	files, err := ioutil.ReadDir(dir)
+func findPlugins(dir string) ([]string, error) {
+	candidates, err := ioutil.ReadDir(dir)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to read plugin directory")
+		return []string{}, errors.Wrapf(err, "couldn't search path %v", dir)
 	}
 
-	for _, file := range files {
-		if filepath.Ext(file.Name()) != ".tmpl" {
-			logrus.WithField("filename", file.Name()).Info("unknown template type")
+	plugins := []string{}
+	for _, candidate := range candidates {
+		if candidate.IsDir() {
 			continue
 		}
-
-		// Read the template file into memory
-		fullPath := path.Join(dir, file.Name())
-		pluginTemplate, err := ioutil.ReadFile(fullPath)
-		if err != nil {
-			return nil, err
-		}
-		dfn, err := loadTemplate(pluginTemplate)
-		if err != nil {
-			logrus.WithError(err).WithField("filename", file.Name()).Info("failed to load plugin")
-			continue
+		ext := filepath.Ext(candidate.Name())
+		if ext == ".yml" || ext == ".yaml" {
+			plugins = append(plugins, filepath.Join(dir, candidate.Name()))
 		}
-		pluginDfns = append(pluginDfns, *dfn)
 	}
+	return plugins, nil
+}
+
+type sonobuoyConfig struct {
+	Driver     string `json:"driver"`
+	PluginName string `json:"plugin-name"`
+	ResultType string `json:"result-type"`
+}
 
-	return pluginDfns, err
+type pluginDefinition struct {
+	SonobuoyConfig sonobuoyConfig   `json:"sonobuoy-config"`
+	Spec           corev1.Container `json:"spec"`
 }
 
-func loadTemplate(tmpl []byte) (*plugin.Definition, error) {
-	t, err := template.New("plugin").Parse(string(tmpl))
+func loadDefinition(file string) (*pluginDefinition, error) {
```

Comment:
optional alternative: I almost want this to take an io.Reader. That way we can move the file concept out into the runner and not need to include testfiles (or mock the fs) when testing

Reply:
We'd still need the testfiles for `findPlugins` I think, but we could certainly do this anyway